automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainGroupRoleSpec.scala

```diff
@@ -0,0 +1,220 @@
+            AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(billingProjectOwner.makeAuthToken())
+          }
+        }
+      }
+    }
+
+    "+ project owner, + group member, create workspace, - project owner" in {
+
+      // a billing project owner, a authdomain group owner and workspace owner
+      val billingProjectOwner = UserPool.chooseProjectOwner
+      implicit val token: AuthToken = billingProjectOwner.makeAuthToken()
+
+      // a billing project owner and a member of authdomain group
+      val student = UserPool.chooseStudent
+
+      withCleanBillingProject(billingProjectOwner, ownerEmails = List(student.email)) { projectName =>
+        withGroup("group", List(student.email)) { groupName =>
+          withWorkspace(projectName, "GroupApiSpec_workspace", Set(groupName)) { workspaceName =>
+
+            AuthDomainMatcher.checkVisibleAndAccessible(projectName, workspaceName, List(groupName))(student.makeAuthToken())
+
+            // remove "student" from billing project
+            Rawls.billing.removeUserFromBillingProject(projectName, student.email, BillingProjectRole.Owner)
+            AuthDomainMatcher.checkNotVisibleNotAccessible(projectName, workspaceName)(student.makeAuthToken())
+          }
+        }
+      }
+    }
+
+    "+ project owner, create workspace, + group member, - group member" in {
+
+      // a billing project owner and a member of the authdomain group
+      val billingProjectOwner = UserPool.chooseProjectOwner
+
+      // a billing project owner, a authdomain group owner and workspace owner
+      val student = UserPool.chooseStudent
+      implicit val token: AuthToken = student.makeAuthToken()
+
+      withCleanBillingProject(billingProjectOwner, List(student.email)) { projectName =>
+        withGroup("group") { groupName =>
+          withWorkspace(projectName, "GroupApiSpec_workspace", Set(groupName)) { workspaceName =>
+
+            AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(billingProjectOwner.makeAuthToken())
+
+            // add a group member "billingProjectOwner" to group
+            Sam.user.addUserToPolicy(groupName, GroupRole.Member.toString, billingProjectOwner.email)
+            register cleanUp Try(Sam.user.removeUserFromPolicy(groupName, GroupRole.Member.toString, billingProjectOwner.email)).recover {
+              case _: RestException =>
+            }
+            AuthDomainMatcher.checkVisibleAndAccessible(projectName, workspaceName, List(groupName))(billingProjectOwner.makeAuthToken())
+
+            // remove a group member "billingProjectOwner" from group
+            Sam.user.removeUserFromPolicy(groupName, GroupRole.Member.toString, billingProjectOwner.email)
+            AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(billingProjectOwner.makeAuthToken())
+          }
+        }
+      }
+    }
+
+    "+ project owner, + group member, - group member" in {
```

Comment:
This test is a duplicate of the previous test and can be removed.

Reply:
👍